http/http/src/main/java/io/helidon/http/ServerResponseHeadersWrapper.java

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.http;
+
+import java.net.URI;
+import java.time.Instant;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalLong;
+import java.util.Spliterator;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+import io.helidon.common.media.type.MediaType;
+
+/**
+ * A response headers wrapper class with rollback capabilities.
+ */
+public class ServerResponseHeadersWrapper implements ServerResponseHeaders {
+
+    private final ServerResponseHeaders delegate;
+    private Function<ServerResponseHeaders, ServerResponseHeaders> rollbackFunction = h -> h;
```

Comment:
(Could just use `Function.identity()`.)

Reply:
(And just a quick sanity check: does `rollbackFunction` need to be `volatile`, given that it is reassigned, or is it guaranteed that will always be on one thread (I assume that must be the case)?)

Reply:
It think it should all be handled by the same thread, but it is worth checking.